### ISSUE TITLE
[DATA] SKW 3370 is changing

### ISSUE THREAD

**Issue by dirkhh** [mrjackwills/adsbdb#17]
**Url of request**
https://api.adsbdb.com/v0/callsign/SKW3370

**Why was the data wrong**
SkyWest changes route on many of their flights fairly frequently. This one is currently used for both directions of Seattle / Rogue Valley, but starting on Thursday it will switch to Seattle Glacier Park

**Additional context**
I really don't know what the best way is to deal with flights like that... especially the increasingly common practice that both directions of a city pair have the same flight number...

**Comment by mrjackwills** [comment]
Yeah, the database is kind of static, it was based on data from a specific point in time, so for flightroute callsigns that change their origin and/or destination, we'd need to work out what the best option would be

**Comment by dirkhh** [comment]
I think there are several separate issues.

One is that right plans change, typically at least twice annually.
The second is that more and more airlines do the weird round-trip with the same flight number.

Those two I have suggestions how to deal with.  This may require some additional changes for the database and the API, though. Once you have that, empower the community to submit changes in an easily digestible format and I think you'll find many people willing to contribute.

What's harder to fix is those weird special flight numbers... And of course the airlines like Southwest that apparently sometimes have three of even four hops with the same flight number...

For those two cases a first suggestion would be to have a special return value that indicates the issue, instead of returning an incorrect route.

But one step at a time, I guess.

**Comment by mrjackwills** [comment]
At the moment I haven't got an easy way to alter data, other than going directly into the database and executing raw SQL queries.

So I think I should code something up, that can run on a regular frequency via a Docker container in the same network. It would read a simple CSV file (`flightroute, Optional<flightroute_icao>, Optional<flightroue_iata>, origin, Optional<midpoint>, destination`) and update the data, as well as clear any cached data. Should be easy enough to create, and then just update the CSV file, based on any submitted GitHub data issues, on a regular basis? On first thoughts, in my mind, that's the most efficient, read laziest, way to do it.

Wondering whether the data issue template should ask for two sources, just to make sure that we input the correct data, and, again, efficient/lazy for me.

As for api changes, it's a v0.x.x release, so I am happy to change the response data, but as you say, would need to work out the best way to convey data about these special, or frequently changing, flights

**Comment by mrjackwills** [comment]
Thanks for the link, yeah sounds like a great opportunity to collaborate. Will spend some time this weekend to look and think about it all.

As for blocking, email me you ip address, and I'll unblock it. I check the logs, maybe weekly, and any abusive looking ip addresses I completely block via nginx.

**Comment by dirkhh** [comment]
done

**Comment by mrjackwills** [comment]
There's now [adsbdb_update](https://github.com/mrjackwills/adsbdb_updater), which should make the task of keeping flight routes up to date painless.

I have already run in against the [input.csv](https://github.com/mrjackwills/adsbdb_updater/blob/main/input.csv) found in the repository

**Comment by dirkhh** [comment]
NICE. I like it.
I am two thirds through a system to identify routes that are likely incorrect, based on adsb info people collect.

This might turn out into a very useful tool to then create such input.csv. I'll keep you posted.

**Comment by mrjackwills** [comment]
The [standing_data](https://github.com/vradarserver/standing-data), at first glance, seems substantially more detailed, a quick SQL query on the adsbdb data, I found 1763 flights that start "DAL", whereas there are [over 3000](https://github.com/vradarserver/standing-data/blob/main/routes/schema-01/D/DAL-all.csv) in that dataset.
